src/ucp/core/ucp_proxy_ep.h

```diff
@@ -24,16 +25,16 @@
  * TODO make sure it works with err handling and print_ucp_info
  */
 typedef struct ucp_proxy_ep {
-    uct_ep_t        super;     /**< Derived from uct_ep */
-    uct_iface_t     iface;     /**< Embedded stub interface */
-    ucp_ep_h        ucp_ep;    /**< Pointer to UCP endpoint */
-    uct_ep_h        uct_ep;    /**< Underlying transport endpoint */
-    int             is_owner;  /**< Is uct_ep owned by this proxy ep */
-    ucp_rsc_index_t rsc_index; /**< Resource index of underlying transport endpoint */
+    uct_ep_t         super;     /**< Derived from uct_ep */
+    uct_base_iface_t iface;     /**< Embedded stub interface */
```

Comment:
we should not use an internal UCT header in UCP
instead, need to check if a uct_ep_t is proxy or regular ep every time before calling uct_ep_connect_to_ep_v2() from UCP

Reply:
hmmm, but test for proxy is UCP call... is it ok?

Reply:
yes; since UCP is always calling uct_ep_connect_to_ep_v2, it should manually check if it's proxy ep or not, every time.